monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/UnfoldSuite.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.internal.operators
+
+import monix.execution.Ack.Continue
+import monix.reactive.{BaseTestSuite, Observable}
+object UnfoldSuite extends BaseTestSuite {
```

Comment:
We could copy few tests from [StateActionObservableSuite](https://github.com/monix/monix/blob/master/monix-reactive/shared/src/test/scala/monix/reactive/internal/builders/StateActionObservableSuite.scala) - especially to do things like:

```scala
val cancelable = Observable
    .fromStateAction(int)(s.clockMonotonic(MILLISECONDS))
    .unsafeSubscribeFn(new Subscriber[Int] {
      implicit val scheduler = s
      def onNext(elem: Int) = {
        sum += 1
        Continue
      }

      def onComplete() = wasCompleted = true
      def onError(ex: Throwable) = wasCompleted = true
})
```

to see if `Observable` completed. I think it could fail currently. Alternatively we can call methods like `sumL` or `completedL` to complete it in nicer way and execute as a `Task`. `subscribe` just processes it asynchronously in the background

Also I'd suggest naming the suite to `UnfoldObservableSuite` - we have also `Iterant` and if we add `unfold` for it then it can be a bit confusing to search for

Reply:
i added the test with `subscribe` for now, I tried to write it like this:
  test("should be cancelable") { implicit s =>

```
    val cancelable = Observable
      .unfold(s.clockMonotonic(MILLISECONDS))(intOption)
        .sumL
        .runToFuture

    cancelable.cancel()

    s.tick()

    assertEquals(cancelable.value, Some(Success(s.executionModel.recommendedBatchSize * 2)))
  }
```

but it returns None - i need to dig deeper and find out why